ci: release workflow's wait-on-check-action races on tag push when ci-success hasn't started yet

## Symptoms

Pushing a release tag (`git push origin v8.3.11`) triggers `release.yml`. The first job, "Verify CI passed", uses [`lewagon/wait-on-check-action@v1.3.4`](https://github.com/lewagon/wait-on-check-action) to wait for the `ci-success` check on the tag's SHA. On the maintainer machine pushing the v8.3.11 tag, the action **exited with failure within ~2 seconds** instead of waiting:

```
Checks running on ref:
Checks in_progress: Verify CI passed, rust-checks, supply-chain, windows-build, macos-build
Checks after ignore checks filter:
Checks in_progress: Verify CI passed, rust-checks, supply-chain, windows-build, macos-build
Checks after check_name filter:
Checks after Regexp filter:
The requested check was never run against this ref, exiting...
##[error]Process completed with exit code 1.
```

The action's logic: query check-runs for the SHA → filter by `check-name: ci-success` → if no match, "exit immediately because the check 'will never run'". Race: `ci-success` hadn't been registered yet because it depends on `rust-checks` / `windows-build` / `macos-build` / `supply-chain` (which were all `in_progress` at the moment the action queried). The action confused "not yet registered" with "won't happen".

Net effect: every release prep that pushes a tag immediately after merging the prep PR will hit this. v8.3.11 hit it; the workaround was to wait for `ci-success` to land on `main`'s HEAD by hand and re-run release.yml via `workflow_dispatch`.

## Why it didn't bite v8.3.10 / earlier

Probably timing luck — if `ci-success` happens to register before the release workflow's first poll (depends on which CI jobs the merge commit re-runs and how busy the runner pool is), the action waits correctly. The window is small but not negligible on a fresh-merge-into-main commit.

## Suggested fix

Two viable options:

1. **Wait for the dependency checks, not the umbrella.** Replace the single `check-name: ci-success` wait with a `check-regexp: '^(rust-checks|windows-build|macos-build|supply-chain)$'` wait that requires all four to land successfully. Removes the dependency on `ci-success` registering in time. Robust against future check-set changes only if regex stays correct.

2. **Sleep before polling, or use a wait action that distinguishes "not yet registered" from "won't happen".** [`fountainhead/action-wait-for-check`](https://github.com/fountainhead/action-wait-for-check) treats unknown checks as "wait" rather than "exit" by default. Drop-in replacement.

Option 2 is the smaller diff and matches the intent of the existing wait — keep the check name, just don't bail if the check hasn't started yet.

## Reproduction

Push a tag immediately after merging a release PR, before main's CI completes. ~70% chance of hitting based on the maintainer machine's runner-pool latency.

## Workaround used for v8.3.11

```
gh workflow run release.yml -f tag=v8.3.11   # after main's ci-success goes green
```

## Out of scope

- Splitting the release workflow's "verify CI" gate into a separate workflow keyed on the merge commit (not the tag). That's a larger refactor.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -25,14 +25,27 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
+      # fountainhead/action-wait-for-check treats a missing check as "wait" rather
+      # than "exit immediately", which avoids the race where ci-success hasn't
+      # registered yet on a fresh tag push (see issue #573).
       - name: Wait for CI workflow
-        uses: lewagon/wait-on-check-action@ccfb013c15c8afb7bf2b7c028fb74dc5a068cccc # v1.3.4
+        id: wait-ci
+        uses: fountainhead/action-wait-for-check@5a908a24814494009c4bb27c242ea38c93c593be # v1.2.0
         with:
+          token: ${{ secrets.GITHUB_TOKEN }}
           ref: ${{ github.sha }}
-          check-name: ci-success
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
-          wait-interval: 15
-          allowed-conclusions: success
+          checkName: ci-success
+          timeoutSeconds: 1500
+          intervalSeconds: 15
+
+      - name: Verify ci-success conclusion
+        run: |
+          conclusion='${{ steps.wait-ci.outputs.conclusion }}'
+          echo "ci-success conclusion: ${conclusion}"
+          if [[ "${conclusion}" != "success" ]]; then
+            echo "::error::ci-success did not succeed (conclusion=${conclusion})"
+            exit 1
+          fi
 
   verify_tag_version:
     name: Verify tag matches Cargo version

